### DISCUSSION TITLE
how to run queries in synchronised method?

### DISCUSSION THREAD

**Discussion by Krzysztof-Lempicki** [tekartik/sqflite#1106] [Q&A] [answered]
Hi,

All methods are `async` here.

My problem is that I run them from `sync` method where I can't use `await`.

My requirements are:
- running in `sync` method
- after running lets say `openDatabase` as a result I need to 
have the created object.  I don't care about time it takes or latency.

How to do this?

**Comment by alextekartik** [comment, marked as answer]
You cannot. You can try sqlite3 which allows calling sqlite in a synchronous way.
